python/ray/train/tests/test_lightning_predictor.py

```diff
@@ -0,0 +1,139 @@
+import re
+import pytest
+import tempfile
+import torch
+
+import numpy as np
+import pytorch_lightning as pl
+from torch.utils.data import DataLoader
+
+import ray
+from ray.air.constants import MAX_REPR_LENGTH, MODEL_KEY
+from ray.air.util.data_batch_conversion import convert_batch_type_to_pandas
+from ray.train.batch_predictor import BatchPredictor
+from ray.train.lightning import LightningCheckpoint, LightningPredictor
+from ray.train.tests.dummy_preprocessor import DummyPreprocessor
+from ray.train.tests.lightning_test_utils import LightningMNISTClassifier
+
+
+def test_repr():
+    model = pl.LightningModule()
+    predictor = LightningPredictor(model)
+
+    representation = repr(predictor)
+
+    assert len(representation) < MAX_REPR_LENGTH
+    pattern = re.compile("^LightningPredictor\\((.*)\\)$")
+    assert pattern.match(representation)
+
+
+def save_checkpoint(model: pl.LightningModule, ckpt_path: str):
+    trainer = pl.Trainer(max_epochs=0)
+    trainer.fit(model, train_dataloaders=DataLoader(torch.randn(1)))
+    trainer.save_checkpoint(ckpt_path)
+
+
+@pytest.mark.parametrize("fs", ["s3", "local"])
+@pytest.mark.parametrize("use_gpu", [True, False])
+@pytest.mark.parametrize("use_preprocessor", [True, False])
+@pytest.mark.parametrize("batch_format", ["numpy", "pandas"])
+def test_predictor(
+    mock_s3_bucket_uri,
+    fs: str,
+    use_preprocessor: bool,
+    use_gpu: bool,
+    batch_format: str,
+):
+    model_config = {
+        "layer_1": 32,
+        "layer_2": 64,
+        "lr": 1e-4,
+    }
+    model = LightningMNISTClassifier(**model_config)
+
+    with tempfile.TemporaryDirectory() as tmpdir:
+        ckpt_path = f"{tmpdir}/{MODEL_KEY}"
+        save_checkpoint(model, ckpt_path)
+
+        # Test load checkpoint from local dir or remote path
+        checkpoint = LightningCheckpoint.from_path(ckpt_path)
+        if fs == "s3":
+            checkpoint.to_uri(mock_s3_bucket_uri)
+            checkpoint = LightningCheckpoint.from_uri(mock_s3_bucket_uri)
```

Comment:
Should we test `from_directory` as well? Aka if they try loading from an AIR checkpoint directory.

Then, we can pytest search over:

```python
"checkpoint_source": ["from_path", "from_uri", "from_directory"]
```